### ISSUE TITLE
Labels wrapped around textarea report label-has-associated-control lint error

### ISSUE THREAD

**Issue by davidwkeith** [jsx-eslint/eslint-plugin-jsx-a11y#639]
**Example**
![image](https://user-images.githubusercontent.com/97965/66343069-b0154800-e8ff-11e9-8480-9d6bdcb838c3.png)

I also see the issue with:

- `<label>foo<meter /></label>`
- `<label>foo<output /></label>`
- `<label>foo<progress /></label>`

**Workaround**
Add the following to your `.eslintrc.js`

```javascript
module.exports = { rules:
    'jsx-a11y/label-has-associated-control': [`error`, {controlComponents:[ 'textarea', 'meter', 'output', 'progress' ]}],
}
```

**Comment by guoyunhe** [comment]
I have same issue with input. No matter it is wrapped input or referenced input.

![image](https://user-images.githubusercontent.com/5836790/68059998-4e31dd80-fd07-11e9-8380-9d4ead2e5722.png)

**Comment by ljharb** [comment]
@guoyunhe you need to *both* wrap and for-ID link, by default.

**Comment by guoyunhe** [comment]
@ljharb but the [document](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/label-has-associated-control.md#succeed-1) says you don't need id when wrapping it:

![image](https://user-images.githubusercontent.com/5836790/68063033-c8b62980-fd16-11e9-8de9-336470dadf6c.png)

**Comment by ljharb** [comment]
Fair point. That seems strange, then (altho i think the default should be "both" not "either")

**Comment by guoyunhe** [comment]
Is here any reason to enforce "both"?

From https://a11yproject.com/checklist/#section-forms

> Use a for/id pairing to guarantee the highest level of browser/assistive technology support.

It doesn't say both.

For screen readers, they can understand either way of label-input connection. Doing both doesn't make it better for end users but add a lot of inconveniences to developers.

**Comment by ljharb** [comment]
Yes - there are some things that only work with nesting (many assistive devices), and others that only work with for-ID linking (Dragon, for example). Doing both gives you 100% a11y coverage, which definitely makes things easier for developers, whose inconvenience is irrelevant compared to users' accessibility :-)

**Comment by davidwkeith** [comment]
@ljharb that is news to me. Then both should be required for all label-able elements. (and a good chance to write a blog post on how to best generate IDs for JSX components without using UUIDs)

**Comment by guoyunhe** [comment]
We need to see evidece. The web is about standard. ID linking is a standard. We shouldn't force developers to support non-standard devices. Wrapping is sometimes impossible if you use some UI library. If it has to be both, there should have been enough evidence in the document.

**Comment by jessebeach** [comment]
I'm going to maintain the option in the rule to account for one, the other,
or both.

**Comment by ljharb** [comment]
@guoyunhe Dragon, specifically, does not support nesting. You're welcome to try it yourself.

Separately, accessibility is *not* about standards, it is about *actual users being able to use websites*, which only depends on what browsers and devices *do* - we *hope* that matches the standard, but if it doesn't, then the standard is irrelevant.

**Comment by guoyunhe** [comment]
The `assert: both` option is actually enforced by `eslint-config-airbnb` package. If you use directly `eslint-plugin-jsx-a11y`, you shouldn't have this issue. But if you use it with `eslint-config-airbnb`, you will have the trouble.

**Comment by ljharb** [comment]
With both the airbnb config, and with using it directly, you can configure the rule however you like - but the goal here is accessibility for *everyone*, and "both" is the only setting that ensures that.

**Comment by jessebeach** [comment]
Thank you all for the report and details. #710 should fix this issue.
